api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1095,7 +1094,9 @@ message Cluster {
 message LoadBalancingPolicy {
   option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.LoadBalancingPolicy";
 
+  // Deprecated and replaced by TypedExtensionConfig.
```

Comment:
Instead of deprecating this whole message, let's deprecate the `name` and `typed_config` fields and add the new `TypedExtensionConfig` field inside of this message.  That way, if we need to add a new field on a per-policy basis in the future, we'll have somewhere to put it.

Reply:
That seems to be an anti-pattern, what field do we expect? In general a per-policy field should be added inside the message packed into Any.

Reply:
If it's a field only for one individual policy, that would make sense.  But if we discover the need to add another field that we want to be set for *all* policies, then this wrapper would provide a place to put that.

Reply:
Could we put that into `CommonLbConfig` then, and include the message in the policy's message? Or is the idea that policies would need to set this field whether they use the common LB config or not?

The verbosity of `load_balancing_policy.policy.typed_extension_config.typed_config.foo` seems a bit excessive. Is it something we could change if/when we actually have such a common field?

Reply:
Not all policies will include `CommonLbConfig` in their config, and the idea here is to have a way to add the field for any policy -- e.g., if we wanted to add something to express the control flow of how the client picks which policy to use.

We could make this change later, but it would be a breaking change at that point.  The idea here is to future-proof a bit.

Reply:
Ack, done.